components/generic/ContentCard.vue

```diff
@@ -50,13 +79,40 @@
         type: String,
         default: 'https://www.europeana.eu/'
       },
-      imageUrl: {
+      image: {
+        type: Object,
+        default: () => {}
+      },
+      creator: {
         type: String,
         default: ''
       },
-      imageTitle: {
+      institution: {
         type: String,
         default: ''
+      },
+      recordId: {
+        type: String,
+        default: ''
+      },
+      recordThumbnailUrl: {
+        type: String,
+        default: ''
+      }
+    },
+    computed: {
+      linkToRecord () {
+        return this.localePath({ name: 'record-all', params: { pathMatch: this.recordId.replace(/^\/+/g, '') } });
+      },
+      backgroundImage () {
+        if (this.recordThumbnailUrl) {
+          return this.recordThumbnailUrl;
+        } else {
+          return (this.image && this.image.fields) ? this.image.fields.file.url : '';
```

Comment:
I think that this component ought only need be aware of an image URL being passed to it, in an `imageUrl` prop. Provision of image URLs from different sources should be handled higher up the rendering hierarchy, perhaps even at plugin or page level.

Reply:
I have moved it one level up to the content card section.